firefox-ios/Client/Frontend/TrackingProtection/CertificatesViewController.swift

```diff
@@ -0,0 +1,280 @@
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        guard !viewModel.certificates.isEmpty else { return 0 }
+        return 4
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: CertificatesCell.cellIdentifier, for: indexPath)
+                as? CertificatesCell else {
+            return UITableViewCell()
+        }
+
+        guard !viewModel.certificates.isEmpty else {
+            return cell
+        }
+
+        let certificate = viewModel.certificates[viewModel.selectedCertificateIndex]
+
+        switch indexPath.row {
+        case CertificatesItemType.subjectName.rawValue:
+            if let commonName = viewModel.getCertificateValues(from: "\(certificate.subject)")[CertificateKeys.commonName] {
+                cell.configure(theme: currentTheme(),
+                               sectionTitle: .Menu.EnhancedTrackingProtection.certificateSubjectName,
+                               items: [(.Menu.EnhancedTrackingProtection.certificateCommonName, commonName)])
+            }
+
+        case CertificatesItemType.issuerName.rawValue:
+            let issuerData = viewModel.getCertificateValues(from: "\(certificate.issuer)")
+            if let country = issuerData[CertificateKeys.country],
+               let organization = issuerData[CertificateKeys.organization],
+               let commonName = issuerData[CertificateKeys.commonName] {
+                cell.configure(theme: currentTheme(),
+                               sectionTitle: .Menu.EnhancedTrackingProtection.certificateIssuerName,
+                               items: [(.Menu.EnhancedTrackingProtection.certificateIssuerCountry, country),
+                                       (.Menu.EnhancedTrackingProtection.certificateIssuerOrganization, organization),
+                                       (.Menu.EnhancedTrackingProtection.certificateCommonName, commonName)])
+            }
+
+        case CertificatesItemType.validity.rawValue:
+            cell.configure(theme: currentTheme(),
+                           sectionTitle: .Menu.EnhancedTrackingProtection.certificateValidity,
+                           items: [
+                            (.Menu.EnhancedTrackingProtection.certificateValidityNotBefore,
+                                certificate.notValidBefore.toRFC822String()),
+                            (.Menu.EnhancedTrackingProtection.certificateValidityNotAfter,
+                                certificate.notValidAfter.toRFC822String())
+                           ])
+
+        case CertificatesItemType.subjectAltName.rawValue:
+            cell.configure(theme: currentTheme(),
+                           sectionTitle: .Menu.EnhancedTrackingProtection.certificateSubjectAltNames,
+                           items: viewModel.getDNSNames(for: certificate))
```

Comment:
Similar comment to line 131.

If `getDNSNames` returns an empty array `[]`, what should happen? I think this probably just makes a cell with only a sectionLabel, which is probably fine since this is unlikely to happen. But might be worth double checking!

Reply:
For websites, without DNSNames, subjectAltName cell will not have values. (only the section label).